tests/client/core/api/test_api.js

```diff
@@ -142,6 +142,36 @@ describe('api', () => {
     });
   });
 
+  describe('featured add-ons api', () => {
+    function mockResponse() {
+      return Promise.resolve({
+        ok: true,
+        json() {
+          return Promise.resolve({
+            results: [
+              { slug: 'foo' },
+              { slug: 'food' },
+              { slug: 'football' },
+            ],
+          });
+        },
+      });
+    }
+
+    it('sets the app, lang, and type query', () => {
+      // FIXME: This shouldn't fail if the args are in a different order.
+      mockWindow.expects('fetch')
+        .withArgs(`${apiHost}/api/v3/addons/featured/?app=android&type=theme&lang=en-US`)
+        .once()
+        .returns(mockResponse());
+      return api.featured({
+        addonType: 'theme',
+        api: { clientApp: 'android', lang: 'en-US' },
+      })
+        .then(() => mockWindow.verify());
```

Comment:
Can you check the value returned here too? As I suggested above I don't think it's returning the list as expected.